src/main/java/org/openrewrite/kotlin/format/SpacesVisitor.java

```diff
@@ -0,0 +1,1232 @@
+            return container.withAfter(container.getAfter().withComments(comments));
+        }
+
+        if (spaceAfter && notSingleSpace(container.getAfter().getWhitespace())) {
+            return container.withAfter(container.getAfter().withWhitespace(" "));
+        } else if (!spaceAfter && onlySpacesAndNotEmpty(container.getAfter().getWhitespace())) {
+            return container.withAfter(container.getAfter().withWhitespace(""));
+        } else {
+            return container;
+        }
+    }
+
+    private static List<Comment> spaceLastCommentSuffix(List<Comment> comments, boolean spaceSuffix) {
+        return ListUtils.mapLast(comments,
+                comment -> spaceSuffix(comment, spaceSuffix));
+    }
+
+    private static Comment spaceSuffix(Comment comment, boolean spaceSuffix) {
+        if (spaceSuffix && notSingleSpace(comment.getSuffix())) {
+            return comment.withSuffix(" ");
+        } else if (!spaceSuffix && onlySpacesAndNotEmpty(comment.getSuffix())) {
+            return comment.withSuffix("");
+        } else {
+            return comment;
+        }
+    }
+
+    /**
+     * Checks if a string only contains spaces or tabs (excluding newline characters).
+     *
+     * @return true if contains spaces or tabs only, or true for empty string.
+     */
+    private static boolean onlySpaces(String s) {
+        for (int i = 0; i < s.length(); i++) {
+            char c = s.charAt(i);
+            if (c != ' ' && c != '\t') {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private static boolean onlySpacesAndNotEmpty(String s) {
+        return !StringUtils.isNullOrEmpty(s) && onlySpaces(s);
+    }
+
+    private static boolean notSingleSpace(String str) {
+        return onlySpaces(str) && !" ".equals(str);
+    }
+
+    // handle space before colon after declaration name
+    private Markers spaceBeforeColonAfterDeclarationName(Markers markers) {
+        return markers.withMarkers(ListUtils.map(markers.getMarkers(), marker -> {
+            if (marker instanceof TypeReferencePrefix) {
+                TypeReferencePrefix mf = (TypeReferencePrefix) marker;
+                return mf.withPrefix(updateSpace(mf.getPrefix(),
+                        style.getOther().getBeforeColonAfterDeclarationName()));
+            }
+            return marker;
+        }));
+    }
+
+    @Override
+    public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, P p) {
+        J.ClassDeclaration c = super.visitClassDeclaration(classDecl, p);
+        if (c.getBody() != null) {
+            c = c.withBody(spaceBefore(c.getBody(), beforeLeftBrace));
+            if (c.getBody().getStatements().isEmpty()) {
+                if (c.getKind() != J.ClassDeclaration.Kind.Type.Enum) {
+                    // Defaulted to `false` in IntelliJ's Kotlin formatting.
+                    boolean withinCodeBraces = false;
```

Comment:
These should be hardcoded, not read from the style

Reply:
Sure, I am going to clean up these kind of hard-coded stuffs after moving all space tests to `rewrite-kotlin`